### ISSUE TITLE
Standard code editor

### ISSUE THREAD

**Issue by thektan** [liferay/liferay-frontend-guidelines#173]
## What is your question?
Is there a standard code editor we should be using for new modules using React in Liferay Portal? Most places are using AUI's Ace Editor ([aui-ace-editor example](https://github.com/liferay/liferay-portal/blob/3eb0efa20b5075c92f71d7dc276f8842cf8686f0/modules/apps/portal-workflow/portal-workflow-web/src/main/resources/META-INF/resources/definition/view_workflow_definition.jsp#L125-L143)), but I also noticed `fragment-web` using CodeMirror in [CodeMirrorEditor.js](https://github.com/liferay/liferay-portal/blob/affce14c75361a8f7ab764b207a47a7cfa31a1fa/modules/apps/fragment/fragment-web/src/main/resources/META-INF/resources/js/fragment-editor/CodeMirrorEditor.js).

cc @oliv-yu @carloslancha

**Comment by carloslancha** [comment]
Hey @thektan AFAIK the idea should be use CodeMirror so we don't increase the usages of AUI stuff.

If you want to use it I guess we can move it to `frontend-js`

cc @jbalsas @wincent

**Comment by jbalsas** [comment]
> If you want to use it I guess we can move it to frontend-js

Yeah... this was bound to happen sooner or later... and we have another place where we might need it (CKEditor Source View).

@thektan, @oliv-yu, could you please share a little bit more of what you're thinking and what you need (ideas, features, mockups...) so we can best advice you?

**Comment by thektan** [comment]
@jbalsas We're working on a search configuration admin section that's basically a GUI for editing JSON. We'll be using a code editor in 2 places.

1. An admin edit form
![2](https://user-images.githubusercontent.com/4496722/89061667-f964d700-d319-11ea-8ed9-42ce422e1280.png)

2. A JSON editor (no mockups yet) similar to Fragment's configuration page
![Screen Shot 2020-07-31 at 10 40 05 AM](https://user-images.githubusercontent.com/4496722/89061833-4779da80-d31a-11ea-9af4-c7819baf656f.png)

**Comment by jbalsas** [comment]
Hey @thektan, those ideas look pretty interesting! What's the roadmap for this? When do you expect to be working on it?

**Comment by thektan** [comment]
@jbalsas We've already started working on the admin edit form. We're tentatively aiming for Dec 2020.

**Comment by jbalsas** [comment]
Hey @thektan, we've been focused more on infra level things we want to get out of the way before the initial GA1, so I'm looking to plan this now for somewhen in Q1 2021. How would that play out for your plans? Maybe you already have this working on your own and we don't really need to push for it 😂

**Comment by thektan** [comment]
Hi @jbalsas! We created our own CodeMirror.js for our module [`portal-search-tuning-blueprints-admin-web`](https://github.com/liferay-search/liferay-portal/blob/LPS-112856_Search_Blueprints/modules/dxp/apps/portal-search-tuning/portal-search-tuning-blueprints-admin-web/src/main/resources/META-INF/resources/js/shared/CodeMirrorEditor.js) similar to the one in [`fragment-web`](https://github.com/liferay/liferay-portal/blob/affce14c75361a8f7ab764b207a47a7cfa31a1fa/modules/apps/fragment/fragment-web/src/main/resources/META-INF/resources/js/fragment-editor/CodeMirrorEditor.js) which has been working for us so far.
